src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/SaveCheckerImplTest.java

```diff
@@ -0,0 +1,195 @@
+    Files.createDirectory(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-20140707T170707S000047-20140707T170707S000048_0" + AVRO_SUFFIX));
+    // Non-chunked file.
+    Files.createDirectory(tmpDir.resolve(SCRIPT_NAME + AVRO_SUFFIX));
+
+    ImmutableMap<String, ChunkCheckpoint> checkpoints = saveChecker.getScriptCheckPoints(tmpDir);
+
+    assertThat(checkpoints).isEqualTo(ImmutableMap.of());
+  }
+
+  @Test
+  public void getScriptCheckpoints_successWithEdgeCaseTimestamp() throws IOException {
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-00000101T000000S000000-00000101T000000S000000_0" + AVRO_SUFFIX));
+    ImmutableMap<String, ChunkCheckpoint> checkpoints = saveChecker.getScriptCheckPoints(tmpDir);
+
+    assertThat(checkpoints)
+        .isEqualTo(
+            ImmutableMap.of(
+                SCRIPT_NAME,
+                ChunkCheckpoint.builder()
+                    .setLastSavedChunkNumber(0)
+                    .setLastSavedInstant(Instant.parse("0000-01-01T00:00:00.000000Z"))
+                    .build()));
+  }
+
+  @Test
+  public void getScriptCheckpoints_unparsableLastTimestamp_throwsException() throws IOException {
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-00000000T000000S000000-00000000T000000S000000_0" + AVRO_SUFFIX));
+
+    IllegalStateException e =
+        assertThrows(IllegalStateException.class, () -> saveChecker.getScriptCheckPoints(tmpDir));
+    assertThat(e).hasMessageThat().contains("Error parsing the last timestamp");
+  }
+
+  @Test
+  public void getScriptCheckpoints_successAsLongAsUnparsableTimestampIsNotInTheLastChunk()
+      throws IOException {
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-00000000T000000S000000-00000000T000000S000000_0" + AVRO_SUFFIX));
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-00000000T000001S000000-00000101T000000S000000_1" + AVRO_SUFFIX));
+
+    ImmutableMap<String, ChunkCheckpoint> checkpoints = saveChecker.getScriptCheckPoints(tmpDir);
+    assertThat(checkpoints)
+        .isEqualTo(
+            ImmutableMap.of(
+                SCRIPT_NAME,
+                ChunkCheckpoint.builder()
+                    .setLastSavedChunkNumber(1)
+                    .setLastSavedInstant(Instant.parse("0000-01-01T00:00:00.000000Z"))
+                    .build()));
+  }
+
+  @Test
+  public void getScriptCheckpoints_nonConsecutiveChunkNumbers_throwsException() throws IOException {
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-20140707T170707S000007-20140707T170707S000008_0" + AVRO_SUFFIX));
+    Files.createFile(
+        tmpDir.resolve(
+            SCRIPT_NAME + "-20140707T170707S000027-20140707T170707S000028_2" + AVRO_SUFFIX));
+
+    IllegalStateException e =
+        assertThrows(IllegalStateException.class, () -> saveChecker.getScriptCheckPoints(tmpDir));
+    assertThat(e).hasMessageThat().contains("consecutiveness");
```

Comment:
I find such a check with only one word to be error prone – what if we have multiple errors message with "consecutiveness" in it? Can we avoided by including full error message in the check